scripts/us_census/pep/annual_population/preprocess.py

```diff
@@ -0,0 +1,862 @@
+    short_forms = _USSTATE_SHORT_FORM
+    data_df[new_col] = data_df[data_col].str.replace(
+        replace_key, "",
+        regex=False).apply(lambda row: short_forms.get(row, row))
+    return data_df
+
+
+def _state_to_geo_id(state: str) -> str:
+    return USSTATE_MAP.get(state, state)
+
+
+def _add_geo_id(data_df: pd.DataFrame, data_col: str,
+                new_col: str) -> pd.DataFrame:
+    data_df[new_col] = data_df[data_col].apply(
+        lambda rec: USSTATE_MAP.get(rec, pd.NA))
+    data_df = data_df.dropna(subset=[new_col])
+    return data_df
+
+
+def _county_to_dcid(geo_id_dict: dict, state_abbr: str, county: str) -> str:
+    """
+    Takes the county name and the state code it's contained in, the dcid of the
+    county is returned.
+    Args:
+        state_abbr: The alpha2 code of the state.
+        county: The county name.
+    Returns:
+        The dcid of the county in DC. Returns an empty string if the county is
+        not found.
+    """
+    if state_abbr in geo_id_dict:
+        if county in geo_id_dict[state_abbr]:
+            return geo_id_dict[state_abbr][county]
+    return county
+
+
+def _process_nationals_1900_1979(ip_file: str, op_file: str) -> None:
+    """
+    Process the nationals data for the year 1900-1979.
+    Args:
+        ip_file (str): Input File Path
+        op_file (str): Output File Path
+    """
+    with open(op_file, 'w', encoding='utf-8') as national_pop_stats:
+        national_pop_stats.write("Year,Location,Count_Person")
+        with open(ip_file, encoding='utf-8') as ipfile:
+
+            for line in ipfile.readlines():
+                # Example:
+                # July 1, 1999      272,690,813          2,442,810        0.90
+
+                if line.startswith(" July 1"):
+                    # Index 9 to 13 provides Year from the line
+                    # and 14 to 30 provides the population count
+                    year = int(line[9:13])
+                    if year >= 1980:
+                        continue
+                    national_pop_stats.write(
+                        "\n" + str(year) + ",country/USA," +
+                        line[14:30].replace(",", "").lstrip().rstrip())
+
+
+def _process_nationals_1980_1989(ip_file: str) -> pd.DataFrame:
+    """
+    Process the nationals data for the year 1980-1989.
+    Args:
+        ip_file (str): Input FIle Path
+
+    Returns:
+        pd.DataFrame: Cleaned DataFrame
+    """
+    usa_rows = ""
+    with open(ip_file, "r", encoding="UTF-8") as raw_file:
+        data = raw_file.readlines()
+        # Extracting below row from the data and clean it.
+        # 00000 United States 226542250 229465744 231664432 233792014 235824908
+        # 00000 United States 237923734 240132831 242288936 244499004 246819222
+        for line in data:
+            if "United States" in line:
+                usa_rows += line.replace("00000", "").strip() + " "
+                usa_cleaned_row = [
```

Comment:
Can this be moved outside the for loop so it processes all usa_rows just once instead of once per row?

Reply:
Moved out of the for loop